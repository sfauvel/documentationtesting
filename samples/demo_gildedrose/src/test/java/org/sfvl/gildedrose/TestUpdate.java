package org.sfvl.gildedrose;

import fr.xebia.katas.gildedrose.Inn;
import fr.xebia.katas.gildedrose.Item;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;


public class TestUpdate {

    @Test
    public void test_update_item() {
        Inn inn = new Inn();
        StringBuffer output = new StringBuffer();
        for (int i = 1; i < 20; i++) {
            for (Item item : inn.getItems()) {
                output.append(String.format("%s,%d,%d\n", item.getName(), item.getSellIn(), item.getQuality()));
            }
            inn.updateQuality();
        }

        Approvals.verify(output);
    }

}
