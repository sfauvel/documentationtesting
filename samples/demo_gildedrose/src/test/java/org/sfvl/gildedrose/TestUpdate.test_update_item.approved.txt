+5 Dexterity Vest,10,20
Aged Brie,2,0
Elixir of the Mongoose,5,7
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,15,20
Conjured Mana Cake,3,6
+5 Dexterity Vest,9,19
Aged Brie,1,1
Elixir of the Mongoose,4,6
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,14,21
Conjured Mana Cake,2,5
+5 Dexterity Vest,8,18
Aged Brie,0,2
Elixir of the Mongoose,3,5
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,13,22
Conjured Mana Cake,1,4
+5 Dexterity Vest,7,17
Aged Brie,-1,4
Elixir of the Mongoose,2,4
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,12,23
Conjured Mana Cake,0,3
+5 Dexterity Vest,6,16
Aged Brie,-2,6
Elixir of the Mongoose,1,3
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,11,24
Conjured Mana Cake,-1,1
+5 Dexterity Vest,5,15
Aged Brie,-3,8
Elixir of the Mongoose,0,2
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,10,25
Conjured Mana Cake,-2,0
+5 Dexterity Vest,4,14
Aged Brie,-4,10
Elixir of the Mongoose,-1,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,9,27
Conjured Mana Cake,-3,0
+5 Dexterity Vest,3,13
Aged Brie,-5,12
Elixir of the Mongoose,-2,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,8,29
Conjured Mana Cake,-4,0
+5 Dexterity Vest,2,12
Aged Brie,-6,14
Elixir of the Mongoose,-3,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,7,31
Conjured Mana Cake,-5,0
+5 Dexterity Vest,1,11
Aged Brie,-7,16
Elixir of the Mongoose,-4,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,6,33
Conjured Mana Cake,-6,0
+5 Dexterity Vest,0,10
Aged Brie,-8,18
Elixir of the Mongoose,-5,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,5,35
Conjured Mana Cake,-7,0
+5 Dexterity Vest,-1,8
Aged Brie,-9,20
Elixir of the Mongoose,-6,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,4,38
Conjured Mana Cake,-8,0
+5 Dexterity Vest,-2,6
Aged Brie,-10,22
Elixir of the Mongoose,-7,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,3,41
Conjured Mana Cake,-9,0
+5 Dexterity Vest,-3,4
Aged Brie,-11,24
Elixir of the Mongoose,-8,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,2,44
Conjured Mana Cake,-10,0
+5 Dexterity Vest,-4,2
Aged Brie,-12,26
Elixir of the Mongoose,-9,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,1,47
Conjured Mana Cake,-11,0
+5 Dexterity Vest,-5,0
Aged Brie,-13,28
Elixir of the Mongoose,-10,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,0,50
Conjured Mana Cake,-12,0
+5 Dexterity Vest,-6,0
Aged Brie,-14,30
Elixir of the Mongoose,-11,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,-1,0
Conjured Mana Cake,-13,0
+5 Dexterity Vest,-7,0
Aged Brie,-15,32
Elixir of the Mongoose,-12,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,-2,0
Conjured Mana Cake,-14,0
+5 Dexterity Vest,-8,0
Aged Brie,-16,34
Elixir of the Mongoose,-13,0
Sulfuras, Hand of Ragnaros,0,80
Backstage passes to a TAFKAL80ETC concert,-3,0
Conjured Mana Cake,-15,0
